src/types/connection/cursor.rs

```diff
@@ -1,60 +1,27 @@
-use crate::{InputValueError, InputValueResult, Result, ScalarType, Value, ID};
-use async_graphql_derive::Scalar;
-use std::ops::{Deref, DerefMut};
+use byteorder::{ReadBytesExt, BE};
+use std::fmt::Display;
 
-/// Cursor scalar
+/// Cursor type
 ///
 /// A custom scalar that serializes as a string.
 /// https://relay.dev/graphql/connections.htm#sec-Cursor
-#[derive(Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug)]
-pub struct Cursor(String);
+pub trait CursorType: Sized {
```

Comment:
I think web should impl `CursorType` for `String` and `ID` as well

Reply:
I agree, but we shouldn't force base64 encoding/decoding on them. There's not a forced rule to implement base64.

I use UUIDs that are "opaque enough" for my system.